sentry_sdk/hub.py

```diff
@@ -479,6 +479,9 @@ def start_span(
             sample_rate = client and client.options["traces_sample_rate"] or 0
             span.sampled = random.random() < sample_rate
 
+        if span.sampled and client:
+            span.init_finished_spans(client.options["max_breadcrumbs"])
```

Comment:
Should probably leave a TODO here as it wont be `max_breadcrumbs` right? Or will it?

Reply:
I would like to couple it to max_breadcrumbs and bump max_breadcrumbs if necessary.

This is a stopgap solution in any case, we should flush more often instead of trim anything. But that's a major refactor of the entire APM product

Reply:
Trimming has been removed again